src/vss_tools/exporters/samm/helpers/vss_helper.py

```diff
@@ -260,14 +260,18 @@ def get_node_description(vss_node: VSSNode) -> str:
 
 
 def has_constraints(vss_node: VSSNode) -> bool:
+    # NOTE: there is a special case for string attributes, which can have a RegExp: pattern field.
+    #       Example: VSS - VehicleIdentification.VIN property
     return (
         hasattr(vss_node.data, "type")
-        and vss_node.data.type in [NodeType.ACTUATOR, NodeType.SENSOR]
```

Comment:
Since the goal seems to be to check against attributes of the `VSSDataDatatype` node type it can probably be simplified to:

```python
data = node.data
if isinstance(data, VSSDataDatatype):
    return data.max or data.min or data.pattern
return False
```
Is `VSSDataProperty` not  being covered on purpose?

Reply:
Thank you.
I will simplify this as advised.

Reply:
Probably need to check them against `None`. So my implementation is not completely correct I suppose. At least the `max` and `min` ;-)

Reply:
Sure, I will try to do my best here :)

Reply:
Hello @sschleemilch , on the question about VSSDataProperty. The code above was a bit obsolete, since I had below check:

```python
vss_node.data.type in [NodeType.ACTUATOR, NodeType.SENSOR, NodeType.ATTRIBUTE]
```
which I guess adds for the VSSDataProperty check. However, since I think that all: sensor, actuator and attribute are children of the VSSDataDatatype, I updated the check as you advised and now it looks like:

```python
def has_constraints(vss_node: VSSNode) -> bool:
    return isinstance(vss_node.data, VSSDataDatatype) and (
        vss_node.data.max or vss_node.data.min or vss_node.data.pattern
    )
```

I tested this on the whole VSS spect and it seems to be fine for every node which has min/max and pattern constraints.

Reply:
Done